SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -210,8 +208,120 @@ public void CreateWorkerWithInvalidTeamIdThrowsPostWorkerException()
+            DatabaseContext.SaveChanges();
+
+            var request = _fixture.Build<UpdateWorkerRequest>()
+                .With(x => x.WorkerId, worker2.Id)
+                .With(x => x.EmailAddress, worker1.Email)
+                .Create();
+
+            Action act = () => _classUnderTest.UpdateWorker(request);
+
+            act.Should().Throw<PatchWorkerException>()
+                .WithMessage($"A worker is already assigned to the email address {request.EmailAddress}");
+        }
+
+        [Test]
+        public void UpdateWorkerUpdatesAnExistingWorkerInDatabase()
+        {
+            var worker = TestHelpers.CreateWorker();
+            var workerTeam = TestHelpers.CreateWorkerTeam(workerId: worker.Id);
+            var team = TestHelpers.CreateTeam(teamId: workerTeam.TeamId);
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+            DatabaseContext.Teams.Add(team);
+            DatabaseContext.SaveChanges();
+
+            var originalWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id).ShallowCopy();
+            originalWorker.Should().BeEquivalentTo(worker);
+
+            var request = TestHelpers.CreateUpdateWorkersRequest(teamId: team.Id, workerId: worker.Id);
+            _classUnderTest.UpdateWorker(request);
+
+            var updatedWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id);
+
+            updatedWorker.Should().NotBeEquivalentTo(originalWorker);
+
+            CompareUpdatedWorkerAndRequest(updatedWorker, request, worker, team);
+        }
+
+        [Test]
+        public void UpdateWorkerIsActiveFlipsDateStartAndDateEndFromTimeToNull()
+        {
+            var worker = TestHelpers.CreateWorker(isActive: true);
+            var workerTeam = TestHelpers.CreateWorkerTeam(workerId: worker.Id);
+            var team = TestHelpers.CreateTeam(teamId: workerTeam.TeamId);
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+            DatabaseContext.Teams.Add(team);
+            DatabaseContext.SaveChanges();
+
+            var originalWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id).ShallowCopy();
+            originalWorker.DateEnd.Should().BeNull();
+            originalWorker.DateStart.Should().NotBeNull();
+
+            var request = TestHelpers.CreateUpdateWorkersRequest(teamId: team.Id, workerId: worker.Id, isActive: false);
+            _classUnderTest.UpdateWorker(request);
+
+            var updatedWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id);
+            updatedWorker.DateStart.Should().BeNull();
+            updatedWorker.DateEnd.Should().NotBeNull();
+        }
+
+        private static void CompareUpdatedWorkerAndRequest(Worker updatedWorker, UpdateWorkerRequest request,
```

Comment:
Suggested renaming to so when we read it in the test, it reads like an expectation.

```suggestion
        private static void UpdatedWorkerShouldBeEquivalentToRequest(Worker updatedWorker, UpdateWorkerRequest request,
```